src/components/explorer/ExplorerSearchResultsTable.js

```diff
@@ -0,0 +1,180 @@
+import React, { useState, useEffect, useMemo } from "react";
+import PropTypes from "prop-types";
+import { useParams } from "react-router-dom";
+import { useSelector, useDispatch } from "react-redux";
+import { Table, Typography, Button, Spin } from "antd";
+
+import SearchSummaryModal from "./SearchSummaryModal";
+import SearchTracksModal from "./SearchTracksModal";
+
+import {
+    setSelectedRows,
+    performIndividualsDownloadCSVIfPossible,
+    performBiosamplesDownloadCSVIfPossible,
+    performExperimentsDownloadCSVIfPossible,
+} from "../../modules/explorer/actions";
+
+const ExplorerSearchResultsTable = ({ data, activeTab, ...props }) => {
+    const { dataset } = useParams();
+    const [currentPage, setCurrentPage] = useState(1);
+    const [pageSize] = useState(25);
+    const [numResults, setNumResults] = useState(0);
+
+    const [summaryModalVisible, setSummaryModalVisible] = useState(false);
+    const [tracksModalVisible, setTracksModalVisible] = useState(false);
+
+    const showingResults = useMemo(() => {
+        const start = numResults > 0 ? currentPage * pageSize - pageSize + 1 : 0;
+        const end = Math.min(currentPage * pageSize, numResults);
+        return `Showing results ${start}-${end} of ${numResults}`;
+    }, [currentPage, pageSize, numResults]);
+
+    const searchResults = useSelector((state) => state.explorer.searchResultsByDatasetID[dataset] || null);
+    const selectedRows = useSelector((state) => state.explorer.selectedRowsByDatasetID[dataset] || []);
+    const isFetchingDownload = useSelector((state) => state.explorer.isFetchingDownload || false);
+    const fetchingSearch = useSelector((state) => state.explorer.fetchingSearchByDatasetID[dataset] || false);
+    const dispatch = useDispatch();
+
+    const handleSetSelectedRows = (...args) => dispatch(setSelectedRows(dataset, ...args));
+
+    const handlePerformIndividualsDownloadCSVIfPossible = (...args) => {
+        if (activeTab === "individuals") {
+            return dispatch(performIndividualsDownloadCSVIfPossible(dataset, ...args));
+        }
+        if (activeTab === "biosamples") {
+            return dispatch(performBiosamplesDownloadCSVIfPossible(dataset, ...args));
+        }
+        if (activeTab === "experiments") {
+            return dispatch(performExperimentsDownloadCSVIfPossible(dataset, ...args));
+        }
+    };
+
+    const onPageChange = (pageObj) => {
+        setCurrentPage(pageObj.current);
+    };
+
+    const tableStyle = {
+        opacity: fetchingSearch ? 0.5 : 1,
+        pointerEvents: fetchingSearch ? "none" : "auto",
+    };
+
+    useEffect(() => {
+        setNumResults(data.length);
+    }, [currentPage, pageSize]);
```

Comment:
I think we can remove this useEffect, the effect function only depends on the `data` prop.